File: python/prime_benchmark.py

import time

def is_prime(n):
    if n <= 1:
        return False
    if n <= 3:
        return True
    if n % 2 == 0 or n % 3 == 0:
        return False
    i = 5
    while i * i <= n:
        if n % i == 0 or n % (i + 2) == 0:
            return False
        i += 6
    return True

def count_primes(limit):
    return sum(1 for i in range(2, limit) if is_prime(i))

if __name__ == "__main__":
    limit = 1000000  # Example limit
    count_primes(limit)  # Warm-up run

    start = time.time()
    count = count_primes(limit)
    end = time.time()

    print(f"Python: {int((end - start) * 1000)} ms")

File: java/PrimeBenchmark.java

import java.util.stream.IntStream;

public class PrimeBenchmark {
    public static boolean isPrime(int num) {
        if (num <= 1) return false;
        if (num <= 3) return true;
        if (num % 2 == 0 || num % 3 == 0) return false;
        for (int i = 5; i * i <= num; i += 6) {
            if (num % i == 0 || num % (i + 2) == 0) return false;
        }
        return true;
    }

    public static long countPrimes(int limit) {
        return IntStream.range(2, limit).filter(PrimeBenchmark::isPrime).count();
    }

    public static void main(String[] args) {
        int limit = 1000000; // Example limit
        countPrimes(limit); // Warm-up run

        long start = System.currentTimeMillis();
        long count = countPrimes(limit);
        long end = System.currentTimeMillis();

        System.out.println("Java: " + (end - start) + " ms");
    }
}
